chapter-06/integration-tests/src/main.rs

```diff
@@ -0,0 +1,76 @@
+use serde::{Deserialize, Serialize};
+
+#[cfg(test)]
+mod tests {
+    use std::env;
+    use aws_sdk_cloudformation::types::Output;
+    use reqwest::{Body, Client};
+    use reqwest::redirect::Policy;
+    use crate::ShortUrl;
+
+    #[tokio::test]
+    async fn when_valid_link_is_passed_should_retrieve_info_and_store() {
+        let api_endpoint = retrieve_api_endpoint().await;
+        
+        let http_client = Client::builder()
+            .timeout(std::time::Duration::from_secs(2))
+            .redirect(Policy::none())
+            .build()
+            .unwrap();
+        
+        let result = http_client.post(format!("{}links", api_endpoint))
+            .header("Content-Type", "application/json")
+            .body(Body::from("{\"url_to_shorten\": \"https://google.com\"}"))
+            .send()
+            .await;
+        
+        assert_eq!(result.is_ok(), true);
+        
+        let response = result.unwrap();
+        
+        assert_eq!(response.status(), 200);
+        
+        let response_data: ShortUrl = serde_json::from_str(response.text().await.unwrap().as_str()).unwrap();
```

Comment:
nitpicking: maybe we can replace the `unwrap`s with `expect`s to have more descriptive error messages in case the test fails/panics